### PR TITLE
Support object vals for mdlc

### PR DESCRIPTION
**Pull request by BrutalCode**
I intend to create PR for #883 
But there are unrelated commits that coming from my fork update process. I hope they will not cause problem.

### CODE REVIEW

File: tests/NLog.UnitTests/Contexts/MappedDiagnosticsLogicalContextTests.cs [hunk 1]
```diff
@@ -45,6 +45,23 @@ public MappedDiagnosticsLogicalContextTests()
...
         [Fact]
+        public void ToSupportObjectVal()
+        {
+            string key = "testKey1";
+            object value = 5;
+
+            MappedDiagnosticsLogicalContext.Set(key, value);
+
+            Assert.Equal(value, MappedDiagnosticsLogicalContext.Get(key));
+        }
```
Comment (304NotModified) [inline tests/NLog.UnitTests/Contexts/MappedDiagnosticsLogicalContextTests.cs]: this should be getObject? (see fail on appveyor (AV))

an unit test on MappedDiagnosticsLogicalContext.Get(key) (result should be "5"), would be also nice

File: src/NLog/MappedDiagnosticsLogicalContext.cs [hunk 2]
```diff
@@ -70,15 +71,34 @@ public static class MappedDiagnosticsLogicalContext
         /// Gets the current logical context named item.
         /// </summary>
         /// <param name="item">Item name.</param>
-        /// <returns>The item value of string.Empty if the value is not present.</returns>
+        /// <returns>The value of <paramref name="item"/>, if defined; otherwise <see cref="String.Empty"/>.</returns>
```
Comment (304NotModified) [inline src/NLog/MappedDiagnosticsLogicalContext.cs]: usage of paramref = +1 :)

File: src/NLog/MappedDiagnosticsLogicalContext.cs [hunk 3]
```diff
@@ -70,15 +71,34 @@ public static class MappedDiagnosticsLogicalContext
...
         public static string Get(string item)
         {
-            string value;
+            return Get(item, null);
+        }
+
+        /// <summary>
+        /// Gets the current logical context named item.
+        /// </summary>
```
Comment (304NotModified) [inline src/NLog/MappedDiagnosticsLogicalContext.cs]: proposal: "Gets the current logical context named item, as `string` "

File: tests/NLog.UnitTests/Contexts/MappedDiagnosticsLogicalContextTests.cs (hunk 1 above)
Comment (BrutalCode) [inline tests/NLog.UnitTests/Contexts/MappedDiagnosticsLogicalContextTests.cs]: Sorry for this silly error :/ I fixed immediately and commit.

File: tests/NLog.UnitTests/Contexts/MappedDiagnosticsLogicalContextTests.cs [hunk 4]
```diff
@@ -45,6 +45,23 @@ public MappedDiagnosticsLogicalContextTests()
...
 
         [Fact]
+        public void ToSupportObjectVal()
+        {
+            string key = "testKey1";
+            object value = 5;
+
+            MappedDiagnosticsLogicalContext.Set(key, value);
+
+            Assert.Equal(value.ToString(), MappedDiagnosticsLogicalContext.Get(key));
```
Comment (304NotModified) [inline tests/NLog.UnitTests/Contexts/MappedDiagnosticsLogicalContextTests.cs]: The tostring isn't good here

Comment (BrutalCode) [inline tests/NLog.UnitTests/Contexts/MappedDiagnosticsLogicalContextTests.cs]: You are right. I just changed it.

File: src/NLog/MappedDiagnosticsLogicalContext.cs (hunk 3 above)
Comment (BrutalCode) [inline src/NLog/MappedDiagnosticsLogicalContext.cs]: Thanks for proposal. I changed.

**Comment by 304NotModified** [comment]
Looks good! It has been merged!

**Comment by 304NotModified** [comment]
Closes #883

**Pull request merged**